core/src/test/java/org/elasticsearch/index/query/MultiMatchQueryBuilderTests.java

```diff
@@ -338,4 +348,56 @@ public void testToFuzzyQuery() throws Exception {
 
         assertEquals(expected, query);
     }
+
+    public void testDefaultField() throws Exception {
+        assumeTrue("test runs only when at least a type is registered", getCurrentTypes().length > 0);
+        QueryShardContext context = createShardContext();
+        MultiMatchQueryBuilder builder = new MultiMatchQueryBuilder("hello");
+        // should pass because we set lenient to true when default field is `*`
+        Query query = builder.toQuery(context);
+        assertThat(query, instanceOf(DisjunctionMaxQuery.class));
+
+        context.getIndexSettings().updateIndexMetaData(
+            newIndexMeta("index", context.getIndexSettings().getSettings(), Settings.builder().putList("index.query.default_field",
+                STRING_FIELD_NAME, STRING_FIELD_NAME_2 + "^5").build())
+        );
+
+        MultiMatchQueryBuilder qb = new MultiMatchQueryBuilder("hello");
+        query = qb.toQuery(context);
+        DisjunctionMaxQuery expected = new DisjunctionMaxQuery(
+            Arrays.asList(
+                new TermQuery(new Term(STRING_FIELD_NAME, "hello")),
+                new BoostQuery(new TermQuery(new Term(STRING_FIELD_NAME_2, "hello")), 5.0f)
+            ), 0.0f
+        );
+        assertEquals(expected, query);
+
+        context.getIndexSettings().updateIndexMetaData(
+            newIndexMeta("index", context.getIndexSettings().getSettings(), Settings.builder().putList("index.query.default_field",
+                STRING_FIELD_NAME, STRING_FIELD_NAME_2 + "^5", INT_FIELD_NAME).build())
+        );
+        // should fail because lenient defaults to false
```

Comment:
Can you add a test here for the `default_field: *` case that it doesn't throw an exception (automatically setting lenient as needed)

Reply:
This is tested by https://github.com/jimczi/elasticsearch/blob/8a130c4e3c6112d889918862221c74d89e153b58/core/src/test/java/org/elasticsearch/index/query/MultiMatchQueryBuilderTests.java#L356